app/models/miq_queue.rb

```diff
@@ -246,6 +246,23 @@ def self.submit_job(options)
     put(options)
   end
 
+  # For services with a dedicated queue_name and worker, such as reporting, high priority work is
+  # run in the 'generic' queue with HIGH_PRIORITY so priority workers or even generic workers can
+  # pick it up immediately. Anything but high priority work or higher is run in the service's
+  # queue, such as 'reporting', and the dedicated worker will process this message following
+  # priority processing.
+  #
+  # Why? Reports are often started by a user who is actively waiting for the result so we need a way
```

Comment:
Since this method is intended to be generic, let's change this text to be an example:

```suggestion
  # Why? For example, with 'reporting', reports are often started by a user who is actively waiting for the result so we need a way
```

(you'll need to re-word wrap this though)